src/modules/socket/server/icmp_socket.cpp

```diff
@@ -37,8 +42,10 @@ icmp_receive_raw(void* arg, raw_pcb*, pbuf* p, const ip_addr_t* addr)
     auto sock = reinterpret_cast<icmp_socket*>(arg);
 
     if (sock->is_filtered(icmp_type(p))) {
-        pbuf_free(p);
-        return (1); /* pbuf was consumed */
+        /* packet not handled by socket so return 0 to allow
+         * icmp_input()/icmp6_input() to handle it
+         */
+        return 0;
```

Comment:
Returning 1 was causing the packets to not get handled by icmp6/ndp so it seems we should return 0 for packets not handled by the socket.